packages/@react-stately/collections/docs/collections.mdx

```diff
@@ -0,0 +1,387 @@
+  {name: 'Snake'}
+]);
+
+<ListBox items={animals} itemKey="name">
+  {item => <Item>{item.name}</Item>}
+</ListBox>
+```
+
+### Why not array map?
+
+If you're an experienced React developer, you may be wondering why we didn't use `animals.map` in this example.
+In fact, you can do this if you want and it will work, but it will not be as performant.
+
+```tsx
+let [animals, setAnimals] = useState([
+  {name: 'Aardvark'},
+  {name: 'Kangaroo'},
+  {name: 'Snake'}
+]);
+
+<ListBox>
+  {animals.map(item => 
+    <Item key={item.name}>{item.name}</Item>
+  )}
+</ListBox>
+```
+
+Using the `items` prop and providing a render function allows React Spectrum to automatically cache the results
+of rendering each item and avoid re-rendering all items in the collection when only one of them changes. This 
+has big performance benefits for large collections.
+
+### Updating data
+
+When you need to update the data to add, remove, or change an item, you can do so using a standard React state update.
+
+**IMPORTANT: all items passed to a collection component must be immutable. Changing a property on an item, or calling 
+`array.push()` or other mutating methods will not work as expected. To make this easier, see the 
+[useListData](useListData.html) hook.**
+
+The following example shows how you might append a new item to the list 
+using <TypeLink links={dataDocs.links} type={dataDocs.exports.useListData} />.
+
+```tsx
+import {useListData} from '@react-stately/data';
+
+let list = useListData({
+  initialItems: [
+    {name: 'Aardvark'},
+    {name: 'Kangaroo'},
+    {name: 'Snake'}
+  ],
+  initialSelectedKeys: ['Kangaroo'],
+  getKey: item => item.name
+});
+
+function addAnimal(name) {
+  list.append({name});
+}
+
+<ListBox items={list.items} itemKey="name">
+  {item => <Item>{item.name}</Item>}
+</ListBox>
+```
+
+### Sections
+
+Sections can be built by returning a &lt;<TypeLink links={docs.links} type={docs.exports.Section} />&gt; instead of
+an &lt;<TypeLink links={docs.links} type={docs.exports.Item} />&gt; item from the top-level item renderer. Sections
+also support an `items` prop with `itemKey` and a renderer function for their children.
+
+```tsx
+let [sections, setSections] = useState([
+  {
+    name: 'People',
+    items: [
+      {name: 'David'},
+      {name: 'Same'},
+      {name: 'Jane'}
+    ]
+  },
+  {
+    name: 'Animals',
+    items: [
+      {name: 'Aardvark'},
+      {name: 'Kangaroo'},
+      {name: 'Snake'}
+    ]
+  }
+]);
+
+<Picker items={sections} itemKey="name">
+  {section =>
+    <Section title={section.name} items={section.items} itemKey="name">
+      {item => <Item>{item.name}</Item>}
+    </Section>
+  }
+</Picker>
+```
+
+When updating nested data, be sure that all parent items change accordingly. Items are immutable,
+so don't use mutating methods like push, or replace a property on a parent item. Instead, copy
+the items that changed as needed. You can do this by hand if you like, but it can be challenging.
```

Comment:
Although totally true, maybe don't call out the fact that this is hard. 
"You can handle this yourself if you require a high level of control. Alternatively you can use the useTreeData hook which will do the heavy lifting for you."